### DISCUSSION TITLE
M2M authentication

### DISCUSSION THREAD

**Discussion by TottePereraSeemli** [FortnoxAB/csharp-api-sdk#284] [General]
Hey guys.

So I'm planning on doing an integration towards Fortnox M2M, but it seems like the only way getting around the Authentication flow is to authenticate as a user - I would rather skip that step since it adds unnecessary complexity to my simple integration.

Does anybody have some more information regarding this? Their support is not very helpful.

**Comment by SWICRIOT** [comment]
@TottePereraSeemli Did you find a solution for this? I'm trying to understand both the Fortnox API and this SDK but I can't make sense of how to authenticate from my console app to my fortnox account without using the user browser login flow.

**Comment by TottePereraSeemli** [comment]
I’ve had a long discussion with Fortnox support about this, I’ve been trying to get them to implement the client credentials flow, but that won’t happen. Their argument is that it is less secure. I dont agree with them, but what to do.

The only way for new integrations to authenticate is the browser login flow that you are referring to.

This project is OS and detached from the Fortnox company.

Sent from Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by adamelfstrom** [comment]
Of note on this topic is that Fortnox previously used the client credential flow for integrations, but changed it a couple of years ago to what we have today. I highly doubt they will consider adding it again after having deprecated it already.
